### PR TITLE
add user resource, with tests

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

https://github.com/OpsLevel/team-platform/issues/308

## Changelog

Changes:
- Add `user` resource, with tests

- [X] List your changes here
- [ ] Make a `changie` entry

## Tophatting

Using this file:
```terraform
# main.tf
resource "opslevel_user" "foo" {
  name  = "Foo Bar"
  email = "foo_bar@baz.com"
  role  = "user"
}
```

Create user. Run `terraform apply`
```terraform
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # opslevel_user.foo will be created
  + resource "opslevel_user" "foo" {
      + email        = "foo_bar@baz.com"
      + id           = (known after apply)
      + last_updated = (known after apply)
      + name         = "Foo Bar"
      + role         = "user"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

opslevel_user.foo: Creating...
opslevel_user.foo: Creation complete after 1s [id=Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
```

Update `main.tf`
```tf
Using this file:
```terraform
# main.tf
resource "opslevel_user" "foo" {
  name  = "Foo Bar the second"
  email = "foo_bar@baz.com"
  role  = "admin"
}
```

Update user. Run `terraform apply`
```tf
opslevel_user.foo: Refreshing state... [id=Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # opslevel_user.foo will be updated in-place
  ~ resource "opslevel_user" "foo" {
        id           = "Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ"
      ~ last_updated = "Friday, 22-Mar-24 14:29:19 CDT" -> (known after apply)
      ~ name         = "Foo Bar" -> "Foo Bar the second"
      ~ role         = "user" -> "admin"
        # (1 unchanged attribute hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

opslevel_user.foo: Modifying... [id=Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ]
opslevel_user.foo: Modifications complete after 1s [id=Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

Remove user. Run `terraform destroy`
```tf
opslevel_user.foo: Refreshing state... [id=Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  # opslevel_user.foo will be destroyed
  - resource "opslevel_user" "foo" {
      - email        = "foo_bar@baz.com" -> null
      - id           = "Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ" -> null
      - last_updated = "Friday, 22-Mar-24 14:30:03 CDT" -> null
      - name         = "Foo Bar the second" -> null
      - role         = "admin" -> null
    }

Plan: 0 to add, 0 to change, 1 to destroy.

Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

opslevel_user.foo: Destroying... [id=Z2lkOi8vb3BzbGV2ZWwvVXNlci8yODMwNQ]
opslevel_user.foo: Destruction complete after 1s

Destroy complete! Resources: 1 destroyed.
```

### CODE REVIEW

File: opslevel/resource_opslevel_user.go [hunk 1]
```diff
@@ -1,123 +1,189 @@
...
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_updated": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"name": schema.StringAttribute{
+				Description: "The name of the user.",
+				Required:    true,
```
Comment (taimoorgit) [inline opslevel/resource_opslevel_user.go:80]:
```suggestion
				Optional:    true,
```

File: tests/resource_user.tftest.hcl [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  assert {
+    condition     = opslevel_user.mock_user.id != null && opslevel_user.mock_user.id != ""
+    error_message = "opslevel_user id should not be empty"
+  }
+
+  assert {
+    condition     = opslevel_user.mock_user.last_updated == "2022-02-24T13:50:07Z"
+    error_message = "wrong last_updated timestamp for opslevel_user"
```
Comment (taimoorgit) [inline tests/resource_user.tftest.hcl:23]: Seems unnecessary to test this

**Review by taimoorgit — CHANGES_REQUESTED** [review]

File: opslevel/resource_opslevel_user.go (hunk 1 above)
Comment (davidbloss) [reply opslevel/resource_opslevel_user.go:80]: The `name` field is required in the current provider - [docs](https://registry.terraform.io/providers/OpsLevel/opslevel/latest/docs/resources/user)

File: tests/resource_user.tftest.hcl (hunk 2 above)
Comment (davidbloss) [reply tests/resource_user.tftest.hcl:23]: It asserts the `last_updated` field exists, which is most of what we're testing at this point

**Review by taimoorgit — APPROVED** [review]

**Pull request merged**